File: frontend/src/types/post.ts
export interface Post {
  postId: number
  categoryId: number
  categoryName: string
  categorySlug: string
  parentPostId: number | null
  authorId: number
  authorUsername: string
  createdAt: string
  updatedAt: string
  title: string | null
  content: string
  isPinned: boolean
  likesCount: number
  replyCount: number
  isLikedByCaller: boolean
  isDeleted: boolean
}

export interface PagedPostsResponse {
  items: Post[]
  page: number
  pageSize: number
  totalCount: number
  totalPages: number
}

export interface CreatePostRequest {
  categoryId: number
  parentPostId?: number | null
  title?: string | null
  content: string
}

export interface UpdatePostRequest {
  title?: string | null
  content: string
}

export interface LikePostResponse {
  postId: number
  likesCount: number
  isLikedByCaller: boolean
}

export interface GetPostsParams {
  categoryId?: number
  parentPostId?: number | null
  page?: number
  pageSize?: number
}

File: UCB-Forum.Server/Models/Post.cs
namespace UCB_Forum.Server.Models;

public class Post
{
    public int PostId { get; set; }
    public int CategoryId { get; set; }
    public int? ParentPostId { get; set; }
    public int AuthorId { get; set; }

    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    public string? Title { get; set; }
    public string Content { get; set; } = string.Empty;
    public bool IsPinned { get; set; }
    public bool IsDeleted { get; set; }
    public int LikesCount { get; set; }

    public Category Category { get; set; } = null!;
    public User Author { get; set; } = null!;
    public Post? ParentPost { get; set; }
    public ICollection<Post> Replies { get; set; } = new List<Post>();
    public ICollection<PostLike> PostLikes { get; set; } = new List<PostLike>();
    public ICollection<Notification> Notifications { get; set; } = new List<Notification>();
}
